nightshift/datastore_transactions.py

```diff
@@ -194,15 +322,57 @@ def retrieve_open_matched_resources_with_full_bib_obtained(
         session.query(Resource)
         .filter(
             Resource.libraryId == libraryId,
+            Resource.resourceCategoryId == resourceCategoryId,
             Resource.status == "open",
-            Resource.deleted == False,
             Resource.fullBib.isnot(None),
         )
         .all()
     )
     return resources
 
 
+def retrieve_processed_files(session: Session, libraryId: int) -> list[str]:
+    """
+    Retrieves file handles of all processed files for specific library.
+
+    Args:
+        session:                `sqlalchemy.Session` instance
+        libraryId:              `nightshift.datastore.Library.nid`
+
+    Returns:
+        list of file handles
+    """
+    instances = session.query(SourceFile).filter_by(libraryId=libraryId).all()
+    return [instance.handle for instance in instances]
+
+
+def set_resources_to_expired(
+    session: Session, resourceCategoryId: int, age: int
+) -> int:
+    """
+    Updates status from 'open' to 'expired' in resources with given category
+    and age in days since bib created in Sierra.
+
+    Args:
+        session:                `sqlalchemy.Session` instance
+        resourceCategoryId:     `nightshift.datastore.ResourceCategory.nid` identifier
+        age:                    number of days since bib created in Sierra
+
+    Returns:
+        number of updated rows in the database
+    """
+    result = session.execute(
+        update(Resource)
+        .where(
+            (Resource.resourceCategoryId == resourceCategoryId)
+            & (Resource.status == "open")
+            & (Resource.bibDate < datetime.utcnow().date() - timedelta(days=age))
+        )
+        .values(status="expired")
+    )
+    return result.rowcount
```

Comment:
Same as my comment above on the delete method. The `query().update()` method is probably a bit cleaner and easier to maintain and also returns the count of rows updated, so might be easier?